### PR TITLE
Add CLI arguments and update spec

### PR DESCRIPTION
**Pull request by jcolicchio**
This PR adds support for the following command line arguments:
`--quiet`
`--use-cache`
`--cache-path`

It also adds support for an upcoming, as-yet unreleased command line argument which will be available in XcodeGen's next release:
`--project-root`

### CODE REVIEW

**Comment by jcolicchio** [comment]
@m-ruhl I understand this repo is two years old but if you're still active please review, hopefully merge and release a version 0.2.0 of your plugin 🤞

File: lib/fastlane/plugin/xcodegen/actions/xcodegen_action.rb [hunk 1]
```diff
@@ -45,6 +48,20 @@ def self.available_options
           FastlaneCore::ConfigItem.new(key: :project,
                                       env_name: "XCODEGEN_PROJECT",
                                       description: "The path to the folder where the project should be generated",
+                                      optional: true),
+          FastlaneCore::ConfigItem.new(key: :quiet,
+                                      env_name: "XCODEGEN_PROJECT",
```
Comment (m-ruhl) [inline lib/fastlane/plugin/xcodegen/actions/xcodegen_action.rb]: You should change the copied env_name

File: lib/fastlane/plugin/xcodegen/actions/xcodegen_action.rb [hunk 2]
```diff
@@ -45,6 +48,20 @@ def self.available_options
...
+                                      description: "Whether to suppress informational and success messages",
+                                      optional: true,
+                                      is_string: false),
+          FastlaneCore::ConfigItem.new(key: :use_cache,
+                                      env_name: "XCODEGEN_PROJECT",
```
Comment (m-ruhl) [inline lib/fastlane/plugin/xcodegen/actions/xcodegen_action.rb]: Also with the other 2 params

**Review by m-ruhl — CHANGES_REQUESTED** [review]
Hi @jcolicchio,
Thanks for the PR. I found only a couple of suggestions..

**Comment by jcolicchio** [comment]
@m-ruhl Done! How does this look? Also, appreciate the quick response :)

**Review by m-ruhl — APPROVED** [review]

**Pull request merged**